Added new "Plugin metadata" section in docs (#9529)
ckeditor/ckeditor5: docs/framework/guides/creating-simple-plugin.md

* {@link framework/guides/tutorials/implementing-a-block-widget Implementing a block widget}
* {@link framework/guides/tutorials/implementing-an-inline-widget Implementing an inline widget}

If you want to read more about the CKEditor 5 architecture, check out the {@link framework/guides/architecture/intro Introduction to CKEditor 5 architecture}.

> If you would like know, how to check what kind of HTML output our plugins return or what UI components are exposed by them, check out the {@link framework/guides/contributing/plugin-metadata plugin metadata} guide.

Reviewer: This will be rather done in this guide - #9414.

In this guide (creating a simple plugin) we should mention here that all 3rd-party CKEditor 5 packages can document their plugins using this meta file. This will allow for easier integrations with their plugins.

Author: I rewrote this sentence - I hope it's better now.

> If you want your plugin to be easily integrated and used by other developers, try to define a metadata file for the package that contains it, as described in the {@link framework/guides/contributing/package-metadata package metadata} guide.

Reviewer: ```suggestion
If you want your plugin to be easily integrated by other developers, learn about {@link framework/guides/contributing/package-metadata package metadata file} that can be added to your package.
```

Author: Done.
